src/data/finset.lean

```diff
@@ -2562,6 +2562,16 @@ lemma disjoint_filter {s : finset α} {p q : α → Prop} [decidable_pred p] [de
     disjoint (s.filter p) (s.filter q) ↔ (∀ x ∈ s, p x → ¬ q x) :=
 by split; simp [disjoint_left] {contextual := tt}
 
+lemma disjoint_filter' {s t : finset α} {p q : α → Prop} [decidable_pred p] [decidable_pred q]
+    (dis : disjoint s t) : disjoint (s.filter p) (t.filter q) :=
+begin
+  simp [disjoint_left],
```

Comment:
This is not the 1-liner you are looking for. But you can use `squeeze_simp` to get rid of the non-terminal simp.